Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp

```diff
@@ -454,15 +454,49 @@ void WebAutomationSession::platformSimulateKeyboardInteraction(WebPageProxy& pag
     }
 #endif // ENABLE(WPE_PLATFORM)
 }
+
+static bool isShiftedChar(UChar keyVal)
+{
+    // https://w3c.github.io/webdriver/#dfn-shifted-character
+    return (keyVal == '~' || keyVal == '|' || keyVal == '{' || keyVal == '}' || keyVal == '<' || keyVal == '>'
+        || keyVal == '!' || keyVal == '@' || keyVal == '#' || keyVal == '$' || keyVal == '%' || keyVal == '^'
+        || keyVal == '&' || keyVal == '*' || keyVal == '(' || keyVal == ')' || keyVal == '_' || keyVal == '+'
+        || (keyVal >= 'A' && keyVal <= 'Z'));
+}
+
 void WebAutomationSession::platformSimulateKeySequence(WebPageProxy& page, const String& keySequence)
 {
     if (page.viewBackend()) {
         platformSimulateKeySequenceLibWPE(page, keySequence, m_currentModifiers);
         return;
     }
 #if ENABLE(WPE_PLATFORM)
-    for (auto codePoint : StringView(keySequence).codePoints())
-        doKeyStrokeEvent(page, true, wpe_unicode_to_keyval(codePoint), m_currentModifiers, true);
+    // https://w3c.github.io/webdriver/#dfn-dispatch-the-events-for-a-typeable-string
+    // For each char of text:
```

Comment:
I appreciate adding comments for what the spec says, but we don't need to comment any line if the code is already easy to understand.